src/Columns/ColumnVector.cpp

```diff
@@ -496,6 +575,14 @@ ColumnPtr ColumnVector<T>::filter(const IColumn::Filter & filt, ssize_t result_s
     static constexpr size_t SIMD_BYTES = 64;
     const UInt8 * filt_end_aligned = filt_pos + size / SIMD_BYTES * SIMD_BYTES;
 
+#if USE_MULTITARGET_CODE
+    if (sizeof(T) == 1 || sizeof(T) == 2 || sizeof(T) == 4 || sizeof(T) == 8)
+    {
+        if (isArchSupported(TargetArch::AVX512VBMI2))
+            TargetSpecific::AVX512VBMI2::doFilterAligned<T, Container, SIMD_BYTES>(filt_pos, filt_end_aligned, data_pos, res_data);
+    }
+#endif
```

Comment:
I guess the only reason why there is not an unintended fallthrough from the AVX512 code to l. 586 is because parameter `filt_pos` (l. 497) is an in-out parameter? It would perhaps be nicer to not rely on such subtleties and structure the code such that both methods run mutually exclusive (--> see my previous comment).